neptune/new/internal/backends/hosted_neptune_backend.py

```diff
@@ -128,54 +112,24 @@ def __init__(self, credentials: Credentials, proxies: Optional[Dict[str, str]] =
             urllib3.disable_warnings()
             ssl_verify = False
 
-        self._http_client = self._create_http_client(ssl_verify, proxies)
-
-        config_api_url = self.credentials.api_url_opt or self.credentials.token_origin_address
-        if proxies is None:
-            verify_host_resolution(config_api_url)
-
-        self._token_http_client = self._create_http_client(ssl_verify, proxies)
-        token_client = create_swagger_client(
-            build_operation_url(config_api_url, self.BACKEND_SWAGGER_PATH),
-            self._token_http_client
+        self._http_client = create_http_client_with_auth(
+            credentials=credentials,
+            ssl_verify=ssl_verify,
+            proxies=proxies
         )
-
-        self._client_config = self._get_client_config(token_client)
-        verify_client_version(self._client_config, neptune_client_version)
-
-        if config_api_url != self._client_config.api_url:
-            token_client = create_swagger_client(
-                build_operation_url(self._client_config.api_url, self.BACKEND_SWAGGER_PATH),
-                self._token_http_client
-            )
-
-        self.backend_client = create_swagger_client(
-            build_operation_url(self._client_config.api_url, self.BACKEND_SWAGGER_PATH),
-            self._http_client
-        )
-        self.leaderboard_client = create_swagger_client(
-            build_operation_url(self._client_config.api_url, self.LEADERBOARD_SWAGGER_PATH),
-            self._http_client
+        self._client_config = get_client_config(
+            credentials=credentials,
+            ssl_verify=ssl_verify,
+            proxies=proxies
         )
 
-        # TODO: Do not use NeptuneAuthenticator from old_neptune. Move it to new package.
-        self._authenticator = NeptuneAuthenticator(
-            self.credentials.api_token,
-            token_client,
-            ssl_verify,
-            proxies)
-        self._http_client.authenticator = self._authenticator
+        verify_client_version(self._client_config, neptune_client_version)
 
-        user_agent = 'neptune-client/{lib_version} ({system}, python {python_version})'.format(
-            lib_version=neptune_client_version,
-            system=platform.platform(),
-            python_version=platform.python_version())
-        self._http_client.session.headers.update({'User-Agent': user_agent})
+        self.backend_client = create_backend_client(self._client_config, self._http_client)
+        self.leaderboard_client = create_leaderboard_client(self._client_config, self._http_client)
 
     def close(self) -> None:
         self._http_client.session.close()
```

Comment:
Why do we close this one?

Reply:
It's only one that we don't store in cache and it's state should be maintained by HostedNeptuneBackend I think

Reply:
For me it looks like it is cached. (hosted_client.py:117)

Reply:
Right, I was thinking about simple `create_http_client`. So it should be dropped here.